scipy/stats/tests/test_multivariate.py

```diff
@@ -363,6 +364,86 @@ def test_frozen_dirichlet(self):
             assert_equal(d.pdf(x[:-1]), dirichlet.pdf(x[:-1], alpha))
             assert_equal(d.logpdf(x[:-1]), dirichlet.logpdf(x[:-1], alpha))
 
+    def test_numpy_rvs_shape_compatibility(self):
+        np.random.seed(2846)
+        alpha = np.array([1.0, 2.0, 3.0])
+        x = np.random.dirichlet(alpha, size=7)
+        assert_equal(x.shape, (7, 3))
+        assert_raises(ValueError, dirichlet.pdf, x, alpha)
+        assert_raises(ValueError, dirichlet.logpdf, x, alpha)
+        dirichlet.pdf(x.T, alpha)
+        dirichlet.pdf(x.T[:-1], alpha)
+        dirichlet.logpdf(x.T, alpha)
+        dirichlet.logpdf(x.T[:-1], alpha)
```

Comment:
I wonder we can still fix this. `dirichlet` is new in scipy 0.15, I wonder how many users it amassed so far. Guessing from a a recent issue where a user was thinking it's actually a multinomial distribution, I guess the answer is "not much yet".  But I'm known to be on a trigger-happy side when it comes to backwards compat, so I'd defer to others on this one. 

If this is not changed, this should be *very* clearly mentioned in the docstring though.